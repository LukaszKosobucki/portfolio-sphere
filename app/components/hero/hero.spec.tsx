import { formatSnapshot } from "@/app/utils/tests/format-snapshot";
import { mockAuthor, mockIntroduction } from "@/app/utils/tests/mocks/hero.mock";
import { render } from "@testing-library/react";
import { describe, expect, test } from "bun:test";
import Hero from "./hero";

describe("Hero Component", () => {
  test("Should match snapshot", () => {
    const { asFragment } = render(<Hero author={mockAuthor} introduction={mockIntroduction} />);

    expect(formatSnapshot(asFragment())).toMatchSnapshot();
  });
});
