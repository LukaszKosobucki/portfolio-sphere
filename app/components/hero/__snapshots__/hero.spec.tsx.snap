// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component Should match snapshot 1`] = `
"<DocumentFragment>
  <section
    class="relative h-screen w-full"
  >
    <div
      class="animate absolute inset-0 top-1/4 overflow-hidden"
      id="planetcont"
    >
      <div
        class="absolute top-0 left-1/2 -translate-x-1/2 w-[250vw] min-h-[100vh] aspect-square rounded-full p-[1px] bg-gradient-to-b from-black/25 dark:from-white/75 from-0% to-transparent to-5%"
        id="crescent"
      >
        <div
          class="w-full h-full bg-white dark:bg-black rounded-full p-[1px] overflow-hidden flex justify-center"
          id="planet"
        >
          <div
            class="w-full h-20 rounded-full bg-neutral-900/25 dark:bg-white/25 blur-3xl"
            id="blur"
          />
        </div>
      </div>
    </div>
    <div
      class="animate absolute h-full w-full flex items-center justify-center"
    >
      <div
        class="relative w-full h-full flex items-center justify-center"
      >
        <div
          class="p-5 text-center"
        >
          <p
            class="animated text-lg md:text-xl lg:text-2xl font-semibold opacity-75"
          >
            Hello, I am ...
          </p>
          <p
            class="animated text-2xl md:text-3xl lg:text-4xl font-bold uppercase text-black dark:text-white"
          >
            Lukasz Kosobucki
          </p>
          <p
            class="animated text-sm md:text-base lg:text-lg opacity-75"
          >
            Frontend Software Engineer, developing creative solutions
          </p>
          <div
            class="animated flex flex-wrap gap-4 justify-center mt-5"
            id="ctaButtons"
          >
            <a
              class="py-2 px-4 rounded truncate text-xs md:text-sm lg:text-base bg-black dark:bg-white text-white dark:text-black hover:opacity-75 blend"
              href="#projects"
            >
              View my projects
            </a>
            <a
              class="py-2 px-4 truncate rounded text-xs md:text-sm lg:text-base border border-black/25 dark:border-white/25 hover:bg-black/5 hover:dark:bg-white/15 blend"
              href="#work"
            >
              View my work
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>"
`;
